docs/docs/Versioning_And_Upgrade/Lua_Toolkit_To_SDK_Plugin_Migration/Plugin_Operations.md

```diff
@@ -0,0 +1,286 @@
+# Plugin Operations
+
+## Summary
+Included below are all plugin operations that have to do with lua migrations. For regular operation information go [here](/References/Plugin_Operations.md).
+
+
+Plugin Operation | **Required** | Decorator | Delphix Engine Operations
+---------------- | -------- | --------- | -------------------------
+[Lua Repository Data Migration](#lua-repository-data-migration) | **No** | `upgrade.repository(lua_version, MigrationType.LUA)` | [Upgrade](/References/Workflows.md#upgrade)
+[Lua Source Config Data Migration](#lua-source-config-data-migration) | **No** | `upgrade.source_config(lua_version, MigrationType.LUA)` | [Upgrade](/References/Workflows.md#upgrade)
+[Lua Linked Source Data Migration](#lua-linked-source-data-migration) | **No** | `upgrade.linked_source(lua_version, MigrationType.LUA)` | [Upgrade](/References/Workflows.md#upgrade)
+[Lua Virtual Source Data Migration](#lua-virtual-source-data-migration) | **No** | `upgrade.virtual_source(lua_version, MigrationType.LUA)` | [Upgrade](/References/Workflows.md#upgrade)
+[Lua Snapshot Data Migration](#snapshot-data-migration) | **No** | `upgrade.snapshot(lua_version, MigrationType.LUA)` | [Upgrade](/References/Workflows.md#upgrade)
+
+
+## Lua Repository Data Migration
+
+A Lua Repository [Data Migration](/References/Glossary.md#data-migration) migrates repository data from an older [schema](/References/Glossary.md#schema) format defined originally from a lua toolkit to an updated schema format.
+
+### Required / Optional
+**Optional.**<br/>
+
+!!! warning
+    You must ensure that all repository data will match your updated repository schema after an upgrade operation. Depending on how your schema has changed, this might imply that you       need to write one or more repository data migrations.
+
+### Delphix Engine Operations
+
+* [Upgrade](/References/Workflows.md#upgrade)
+
+### Signature
+
+`def migrate_repository(old_repository)`
+
+### Decorator
+
+`upgrade.repository(lua_version, MigrationType.LUA)`
+
+### Decorator Arguments
+
+Argument | Type | Description
+-------- | ---- | -----------
+lua_version | String | The lua version of the migration that the upgrade would be migrating from. This is the ID of this migration. The version here is actually just the major and minor version of the lua toolkit therefore each defined migration's lua_version per repository data migration must be unique.
+migration_type | String | This migration type field indicates whether the operation is lua or just a regular data migration. Signify this as LUA so that the major minor version can be used. If not defined, this operation will default to a regular [repository data migration](/References/Plugin_Operations.md#repository-data-migration).
```

Comment:
hmm not that I can think of. since sometimes they'll go directly to this this specific section (ie repo or source config or etc etc) so it makes sense that it's copied? I also think that in the non-lua pages there's a lot of copy paste too and I think it's okay? if you feel strongly we can brainstorm how we'd fix this?